### PR TITLE
Add unit test for JobStoreCenter

### PR DESCRIPTION
**Pull request by andreeanica16**
1. Modify pom.xml to include all the needed dependencies
2. Modify how the ProjectLoaderStub class deals with an update. The update is added by calling the method setUpdate, that takes 2 parameters: the first parameter is a list with jobs, and only the fields that have been modified since our last call would have a value different from null. The other parameter is the list with the job in its current form, after being modified. These jobs would replace the initial ones, and so the ProjectLoaderStub would return the updated version of the job when fetching the job again.
3. Added unit test for JobStoreCenter that tests storing the information correctly to Datastore

### CODE REVIEW

**Comment by andreeanica16** [comment]
@iht @juliocc @tblanshard

File: src/test/java/com/google/sps/JobStoreCenterTest.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    // Add a project that has only one job (the job is finalised)
+    Map<String, List<JobJSON>> mapOfJobs = new HashMap<>();
+    mapOfJobs.put("europe-west1", Arrays.asList(JOB1));
+    mapOfJobs.put("europe-west2", Arrays.asList(JOB2));
+
+    ProjectLoader projectLoader = new ProjectLoaderStub(mapOfJobs, PROJECT1);
+    clock.setTime(TIME2);
+    jobStoreCenter.dealWithProject(PROJECT1, projectLoader);
+    
+    // Datastore should contain only 2 elements
```
Comment (tblanshard) [inline src/test/java/com/google/sps/JobStoreCenterTest.java:200]: I think this is an old comment/needs updating?

**Review by tblanshard — APPROVED** [review]
LGTM :) Only one really tiny change I think for a comment line

File: src/test/java/com/google/sps/JobStoreCenterTest.java (hunk 1 above)
Comment (andreeanica16) [reply src/test/java/com/google/sps/JobStoreCenterTest.java:200]: Yes, you are right. Thanks :D

Comment (andreeanica16) [reply src/test/java/com/google/sps/JobStoreCenterTest.java]: Deleted in https://github.com/googleinterns/step229-2020/pull/67/commits/976407ef6850b720819d5c8fe5f3c54436ec838f

**Review by tblanshard — APPROVED** [review]
LGTM :)

**Pull request merged**